### ISSUE TITLE
C#7 throw expressions result in SA0102

### ISSUE THREAD

**Issue by PeteCooney** [StyleCop/StyleCop#92]
When using the new C#7 throw expressions StyleCop returns a false-positive instance of SA0102 (syntax error) and stops processing the rest of the file.

For example:
```
this.LocalProperty = parameterValue ?? throw new ArgumentNullException(nameof(parameterValue), $"A value must be provided for {nameof(parameterValue)}");
```
Verified on Visual Studio 2017 Enterprise using StyleCop 5.0.6187.1

**Comment by msmolka** [comment]
Also inline is assigning the same. 
How to disable this rule?
